src/components/forms/JurisdictionHierachyDownload/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import Papaparse from 'papaparse';
+import React, { useState } from 'react';
+import { Button, FormGroup } from 'reactstrap';
+import JurisdictionSelect from '../../../components/forms/JurisdictionSelect';
+import {
+  DOWNLOAD,
+  DOWNLOAD_FILE,
+  DOWNLOADING,
+  FILE,
+  FILE_DOWNLOADED_SUCCESSFULLY,
+  JURISDICTION_HIERARCHY_TEMPLATE,
+  SELECT_COUNTRY,
+} from '../../../configs/lang';
+import { OPENSRP_V2_SETTINGS, TEXT_CSV } from '../../../constants';
+import { downloadFile, successGrowl } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+import { RawOpenSRPHierarchy, TreeNode } from '../../../store/ducks/opensrp/hierarchies/types';
+import { generateJurisdictionTree } from '../../../store/ducks/opensrp/hierarchies/utils';
+import { LOCATION } from '../../TreeWalker/constants';
+
+export interface JurisdictionHierachyFile {
+  jurisdiction_id: string;
+  jurisdiction_name: string;
+}
+
+/** interface for each select dropdown option */
+export interface Option {
+  id: string;
+  name: string;
+}
+
+/** interface for each select dropdown option */
+export interface JurisdictionHierachyDownloadFormFields {
+  jurisdictions: Option;
+}
+
+/**
+ * interface for the Jurisdiction hierarchy download form fields
+ */
+export interface JurisdictionHierachyDownloadFormProps {
+  disabledFields: string[];
+  serviceClass: OpenSRPService;
+  submitForm: (
+    setSubmitting: (isSubmitting: boolean) => void,
+    setGlobalError: (errorMessage: string) => void,
+    props: JurisdictionHierachyDownloadFormProps,
+    values: JurisdictionHierachyDownloadFormFields
+  ) => void;
```

Comment:
lets declare this type somwhere else other than inline here. i mean like a variable and then pass it